libs/arrow-meta/src/main/kotlin/arrow/meta/phases/codegen/ir/IrUtils.kt

```diff
@@ -289,3 +284,17 @@ fun IrReturnTarget.returnType(irBuiltIns: IrBuiltIns) =
     is IrReturnableBlock -> type
     else -> error("Unknown ReturnTarget: $this")
   }
+
+// copied from org.jetbrains.kotlin.ir.interpreter
+fun IrClass.internalName(): String {
```

Comment:
We could delegate to the original impl if it was internal by bypassing the internal modifier.

Since this is a jvm project it will always be public in the binary. That way we could avoid code duplication but also prevent relying on the hack annotation everywhere.

(Need to look up the annotations)

Reply:
That is really out of my knowledge. Do you have a pointer or an example where this is done? Because using the original implementation is clearly better than keeping this one in sync with the compiler.

Reply:
`@file:Suppress("UNUSED", "INVISIBLE_MEMBER", "INVISIBLE_REFERENCE")` you need to add this to the file, which will allow you to import and call the `internal` function from the Kotlin Compiler.

It needs to be on the file because otherwise the `import xxxx` will not resolve.

Reply:
Example usage:
 
 - https://github.com/Kotlin/kotlinx.coroutines/blob/master/kotlinx-coroutines-debug/src/DebugProbes.kt#L5
 
 
 They're accessing `DebugProbesImpl` from the core module.

Reply:
Done!